### ISSUE TITLE
Inconsistent error with podman build using --secret

### ISSUE THREAD

**Issue by Aetylus** [containers/podman#17899]
### Issue Description

Attempting to build an image using `podman build` along with a secret using `--secret` will inconsistently return the following error:
```
time="2023-03-23T09:59:08-04:00" level=error msg="1 error occurred:\n\t* archive/tar: write too long\n\n\n"
Error: Post "http://d/v4.4.2/libpod/build?dockerfile=%5B%22Containerfile%22%5D&forcerm=1&httpproxy=1&identitylabel=1&idmappingoptions=%7B%22HostUIDMapping%22%3Atrue%2C%22HostGIDMapping%22%3Atrue%2C%22UIDMap%22%3A%5B%5D%2C%22GIDMap%22%3A%5B%5D%2C%22AutoUserNs%22%3Afalse%2C%22AutoUserNsOpts%22%3A%7B%22Size%22%3A0%2C%22InitialSize%22%3A0%2C%22PasswdFile%22%3A%22%22%2C%22GroupFile%22%3A%22%22%2C%22AdditionalUIDMappings%22%3Anull%2C%22AdditionalGIDMappings%22%3Anull%7D%7D&isolation=1&jobs=1&layers=1&networkmode=0&nsoptions=%5B%7B%22Name%22%3A%22user%22%2C%22Host%22%3Atrue%2C%22Path%22%3A%22%22%7D%5D&omithistory=0&output=testimage&outputformat=application%2Fvnd.oci.image.manifest.v1%2Bjson&platform=%2F&pullpolicy=missing&rm=1&secrets=%5B%22id%3Dmysecret%2Csrc%3Dpodman-build-secret3109220150%22%5D&shmsize=67108864&t=testimage": io: read/write on closed pipe
```
This only occurs some of the time. Running the exact command multiple times will occasionally build the image without any error.

### Steps to reproduce the issue

Containerfile:
```dockerfile
FROM ubuntu:23.04

RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
```

mysecret.txt:
```
secretpassword
```

Run:
```sh
$  podman build --secret id=mysecret,src=mysecret.txt -t testimage .
```

### Describe the results you received

The `podman build` command will occasionally return the following error:
```
time="2023-03-23T09:59:08-04:00" level=error msg="1 error occurred:\n\t* archive/tar: write too long\n\n\n"
Error: Post "http://d/v4.4.2/libpod/build?dockerfile=%5B%22Containerfile%22%5D&forcerm=1&httpproxy=1&identitylabel=1&idmappingoptions=%7B%22HostUIDMapping%22%3Atrue%2C%22HostGIDMapping%22%3Atrue%2C%22UIDMap%22%3A%5B%5D%2C%22GIDMap%22%3A%5B%5D%2C%22AutoUserNs%22%3Afalse%2C%22AutoUserNsOpts%22%3A%7B%22Size%22%3A0%2C%22InitialSize%22%3A0%2C%22PasswdFile%22%3A%22%22%2C%22GroupFile%22%3A%22%22%2C%22AdditionalUIDMappings%22%3Anull%2C%22AdditionalGIDMappings%22%3Anull%7D%7D&isolation=1&jobs=1&layers=1&networkmode=0&nsoptions=%5B%7B%22Name%22%3A%22user%22%2C%22Host%22%3Atrue%2C%22Path%22%3A%22%22%7D%5D&omithistory=0&output=testimage&outputformat=application%2Fvnd.oci.image.manifest.v1%2Bjson&platform=%2F&pullpolicy=missing&rm=1&secrets=%5B%22id%3Dmysecret%2Csrc%3Dpodman-build-secret3109220150%22%5D&shmsize=67108864&t=testimage": io: read/write on closed pipe
```

### Describe the results you expected

The same `podman build` command will alternatively run successfully:
```
STEP 1/2: FROM ubuntu:23.04
STEP 2/2: RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
secretpassword
COMMIT testimage
--> Pushing cache []:319b70baa4bd06570e3c7fdf2f225fd3293ee98170ccb5272dce831c6bd34ee5
--> dc3a689b5c0
Successfully tagged localhost/testimage:latest
```

### podman info output

```yaml
host:
  arch: amd64
  buildahVersion: 1.29.0
  cgroupControllers: []
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: conmon-2.1.5-1.fc36.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.5, commit: '
  cpuUtilization:
    idlePercent: 99.88
    systemPercent: 0.09
    userPercent: 0.03
  cpus: 8
  distribution:
    distribution: fedora
    variant: container
    version: "36"
  eventLogger: journald
  hostname: Priscilla
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.10.102.1-microsoft-standard-WSL2
  linkmode: dynamic
  logDriver: journald
  memFree: 19203522560
  memTotal: 20081319936
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: crun-1.8.1-1.fc36.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.8.1
      commit: f8a096be060b22ccd3d5f3ebe44108517fbf6c30
      rundir: /run/user/1000/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +WASM:wasmedge +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-0.2.beta.0.fc36.x86_64
    version: |-
      slirp4netns version 1.2.0-beta.0
      commit: 477db14a24ff1a3de3a705e51ca2c4c1fe3dda64
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 5368709120
  swapTotal: 5368709120
  uptime: 2h 3m 22.00s (Approximately 0.08 days)
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - docker.io
store:
  configFile: /home/user/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/user/.local/share/containers/storage
  graphRootAllocated: 269490393088
  graphRootUsed: 2134208512
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 28
  runRoot: /run/user/1000/containers
  transientStore: false
  volumePath: /home/user/.local/share/containers/storage/volumes
version:
  APIVersion: 4.4.1
  Built: 1676629882
  BuiltTime: Fri Feb 17 05:31:22 2023
  GitCommit: ""
  GoVersion: go1.18.10
  Os: linux
  OsArch: linux/amd64
  Version: 4.4.1
```

### Podman in a container

No

### Privileged Or Rootless

None

### Upstream Latest Release

No

### Additional environment details

This is a Windows 10 installation of Podman installed using `winget install RedHat.Podman`.

**Comment by flouthoc** [comment]
I'll check this thanks.

**Comment by flouthoc** [comment]
@Aetylus Is content in `secret.txt` too large ? or how big is your `contextdir` ?

**Comment by Aetylus** [comment]
> @Aetylus Is content in `secret.txt` too large ? or how big is your `contextdir` ?

It shouldn't be, as it's 15 characters long encoded in UTF-8, for a total of 16 bytes (carriage return included).

As far as the `contextdir` goes, do you mean the total size of all files in the build directory? If so, it's just the two files, the `Containerfile` and `secret.txt` for a total of 100 bytes even. 16 for `secret.txt` and 84 for `Containerfile`.

**Comment by flouthoc** [comment]
@Aetylus I tried it running in a loop for continuously for more than 20 times and error did not occur to me, it seems issue could have been with the old version, latest upstream does not have this issue `4.6.0-dev`, I'd request you to update both `podman-remote` client and server to the latest released version. I'm closing this issue but please re-open if you think if this persists in the recent version as well.

My runs
```console
./run.sh 
STEP 1/2: FROM ubuntu:23.04
STEP 2/2: RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
secretpassword
COMMIT testimage
--> e9535fb307b6
Successfully tagged localhost/testimage:latest
e9535fb307b6bb5438a080d205ba2ee41379a318656678a5fc1cc7d0bc4164ac
STEP 1/2: FROM ubuntu:23.04
STEP 2/2: RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
secretpassword
COMMIT testimage
--> f18341670a77
Successfully tagged localhost/testimage:latest
f18341670a776698eaa422c4aa0f45a15a4c78a699d957aa0f9b437098d2c4ac
STEP 1/2: FROM ubuntu:23.04
STEP 2/2: RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
secretpassword
COMMIT testimage
--> dd3e8937b7f5
Successfully tagged localhost/testimage:latest
dd3e8937b7f5ef94e536a56cce54365b3c8dad78b5d4d73570f93c7ba9ee809d
STEP 1/2: FROM ubuntu:23.04
STEP 2/2: RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
secretpassword
COMMIT testimage
--> 419de460510a
Successfully tagged localhost/testimage:latest
419de460510ad543978c849b3180194e0332257d8dca835f51b308b287a8fcdc
STEP 1/2: FROM ubuntu:23.04
STEP 2/2: RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
secretpassword
COMMIT testimage
--> 53635d14f457
Successfully tagged localhost/testimage:latest
53635d14f4579b57991df26a2380048697265b8eda75d34015d69d09f23b1132
STEP 1/2: FROM ubuntu:23.04
STEP 2/2: RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
secretpassword
COMMIT testimage
--> ab435cc15a75
Successfully tagged localhost/testimage:latest
ab435cc15a7565e46217000d38152b2490a3b537af29c4d1296f9afc92344e87
STEP 1/2: FROM ubuntu:23.04
STEP 2/2: RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
secretpassword
COMMIT testimage
--> 529c43667218
Successfully tagged localhost/testimage:latest
529c43667218cabb41b120d35b33cfacc39e39a2f3c9199d2558b4318c474133
STEP 1/2: FROM ubuntu:23.04
STEP 2/2: RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
secretpassword
COMMIT testimage
--> 9cdafafa73a5
Successfully tagged localhost/testimage:latest
9cdafafa73a5a30e857e146157b9df5fdd4b0c37a2766f1cf5b345084543625a
STEP 1/2: FROM ubuntu:23.04
STEP 2/2: RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
secretpassword
COMMIT testimage
--> ccbc94da2df2
Successfully tagged localhost/testimage:latest
ccbc94da2df27230078c2988c2ac47b67f8c4644816381ffd8ea4d92e2901d6d
STEP 1/2: FROM ubuntu:23.04
STEP 2/2: RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
secretpassword
COMMIT testimage
--> 2d95052ab6b3
Successfully tagged localhost/testimage:latest
2d95052ab6b3fc98bcc703fcaea453853a060c7a38abd9c88643fe4afb137eb2
STEP 1/2: FROM ubuntu:23.04
STEP 2/2: RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
secretpassword
COMMIT testimage
--> 7f30587aa485
Successfully tagged localhost/testimage:latest
7f30587aa485337b39ad6c00714516a6a6153a8c66808ff2cc5a5c4614f49dd1
STEP 1/2: FROM ubuntu:23.04
STEP 2/2: RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
secretpassword
COMMIT testimage
--> 8bb39f2aa456
Successfully tagged localhost/testimage:latest
8bb39f2aa456f3eee8487c50b3b8fbf87da585b746cc8cfea503328c1ad41f73
STEP 1/2: FROM ubuntu:23.04
STEP 2/2: RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
secretpassword
COMMIT testimage
--> 2023c835a177
Successfully tagged localhost/testimage:latest
2023c835a1774338a1735dd497f0d68065e20ba49428f5da37ea10ce54b97bbb
STEP 1/2: FROM ubuntu:23.04
STEP 2/2: RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
secretpassword
COMMIT testimage
--> c2787a7c2f67
Successfully tagged localhost/testimage:latest
c2787a7c2f672c376ae8586eb77061824f40037f9e067db552c5df4ce2aaecce
STEP 1/2: FROM ubuntu:23.04
STEP 2/2: RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
secretpassword
COMMIT testimage
--> 1d3076132164
Successfully tagged localhost/testimage:latest
1d30761321647802bbab9288ffaf81a0e453d9890d3bbaab5aee314c85842269
STEP 1/2: FROM ubuntu:23.04
STEP 2/2: RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
secretpassword
COMMIT testimage
--> 1b3a8981354f
Successfully tagged localhost/testimage:latest
1b3a8981354faa888fa60f334c9599b5573da77f088f304aec9c90a24263ead2
STEP 1/2: FROM ubuntu:23.04
STEP 2/2: RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
secretpassword
COMMIT testimage
--> 5b5e8a202af6
Successfully tagged localhost/testimage:latest
5b5e8a202af60e4dc0353eee797e328797fc959e3ad54d4bc8d409447ac0fb13
STEP 1/2: FROM ubuntu:23.04
STEP 2/2: RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
secretpassword
COMMIT testimage
--> bce3b543ed92
Successfully tagged localhost/testimage:latest
bce3b543ed9206849f639e8045b3118f6455bdc87bae7fb7a7cae86d6324cd9c
STEP 1/2: FROM ubuntu:23.04
STEP 2/2: RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
secretpassword
COMMIT testimage
--> 3d51c37c9871
Successfully tagged localhost/testimage:latest
3d51c37c987140684343c4a9f8a5419a0b45d5c03189c31ca11d6e11500e7c59
STEP 1/2: FROM ubuntu:23.04
STEP 2/2: RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
secretpassword
COMMIT testimage
--> 96703d9ddadf
Successfully tagged localhost/testimage:latest
96703d9ddadfcf11fb1279ac97d15828d193f95ff4646484db5f8dda1e8e4439
```

**Comment by j2ghz** [comment]
> I tried it running in a loop for continuously for more than 20 times and error did not occur to me, it seems issue could have been with the old version, latest upstream does not have this issue `4.6.0-dev`, I'd request you to update both `podman-remote` client and server to the latest released version. I'm closing this issue but please re-open if you think if this persists in the recent version as well.
> \- @flouthoc

I tested on three systems:

`Windows 11 Pro (22621.2428) podman 4.7.1 + podman machine in WSL2`
`Windows 11 Enterprise (22621.2428) podman 4.6.2 + podman machine in WSL2`
These two machines will fail the first build, and the rest will succeed. I tried a few minutes later, and it still succeeds. But trying an hour or two later (I wasn't timing it) again triggers 1 failure, and further retries succeed.

`Windows Server 2022 Datacenter Azure Edition (20348.2031) podman 4.7.1 + podman machine in WSL2`
Always fails, no matter how often it's run.

I tested it with the `Steps to reproduce the issue` from above. It looks like it's system dependent, but I'm not sure what differences to looks for.
